### PR TITLE
DEV/DOC: Adding PPM classes for power meter that uses responsivity in…

### PR DESCRIPTION
**Pull request by KaushikMalapati**
… calculations

## Description

Adding two classes - PPM_R_PowerMeter and PPM_R to be used with power meters that use new way of calculating pulse energies using responsivity and support automatic collection of background voltages.

## Motivation and Context

https://jira.slac.stanford.edu/browse/ECS-875
New classes to support changes from this ticket

## How Has This Been Tested?

Not yet tested

## Where Has This Been Documented?

New pre-release notes and docstrings on classes

## Pre-merge checklist
- [ ] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [ ] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [ ] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Review by ZLLentz — CHANGES_REQUESTED** [review]
Can you make this the normal PPM class instead of creating a new one?

Counterintuitively, creating a class bifurcation causes more backwards-compatibility problems rather than preventing them, because it requires us to update `happi` independently for each instance, and then anyone running an older version of pcdsdevices suddenly cannot `happi` load the updated PPM because the new class doesn't exist yet!

Rather than have some clients that can't use the PPMs at all, I'd prefer some distributed partial PV disconnects (provided they don't conflict with other PPM features)

I realize this is a fundamental issue with how we deploy `happi` devices and I'm open to suggestions on ways to do this better.

**Comment by ZLLentz** [comment]
don't worry about the various weird test failures- there are some cascading issues related to the numpy update

File: docs/source/upcoming_release_notes/1249-Responsive_PPM.rst
Comment (ZLLentz) [inline docs/source/upcoming_release_notes/1249-Responsive_PPM.rst:1]: Looks like we have two copies of the pre-release notes in this pr

File: pcdsdevices/pim.py
Comment (ZLLentz) [inline pcdsdevices/pim.py:447]: btw, `command-enum` makes the typhos screen display separate buttons for each option, arranged vertically. Just making sure you know, some people prefer the combobox but it's a matter of taste and I don't have a strong opinion.

**Review by ZLLentz — COMMENTED** [review]
The class as-designed is really good

File: pcdsdevices/pim.py
Comment (KaushikMalapati) [reply pcdsdevices/pim.py:447]: I was assuming that you needed command-enum for the combobox.

Comment (ZLLentz) [reply pcdsdevices/pim.py]: These aren't documented super well, the only way to know is to flip through the decorators at https://github.com/pcdshub/typhos/blob/master/typhos/widgets.py

**Comment by ZLLentz** [comment]
I think this looks great! Let's hold until we can try this on the real thing and then merge as-is if we're happy with it.

**Comment by tangkong** [comment]
This also needs a master branch merge, but I agree with Zach's comments above

**Review by ZLLentz — APPROVED** [review]
I've confirmed that the screen is still totally functional for PPMs that haven't been upgraded yet :+1:

I'd like to see an example of it for an already updated device/ioc but this can wait until we have one.

**Comment by ZLLentz** [comment]
I think this is still good as-is to merge but I need to click update branch first
To avoid clicking it twice I'm going to click it after the other PR is merged
And then we can merge this one after the tests re-pass again

**Pull request merged**
